src/client/dfs/dfs.c

```diff
@@ -3046,7 +3049,7 @@ dfs_read_int(dfs_t *dfs, dfs_obj_t *obj, daos_off_t off, dfs_iod_t *iod,
 	daos_task_set_priv(task, params);
 	rc = tse_task_register_cbs(task, NULL, 0, 0, read_cb, NULL, 0);
 	if (rc)
-		D_GOTO(err_params, rc);
+		D_GOTO(err_params, daos_der2errno(rc));
```

Comment:
Hmm. rc will not be assigned with correct error here. It should be following:
`D_GOTO(err_params, rc = daos_der2errno(rc));`

Reply:
yes indeed. i will push a follow on PR